sections/security/secretmanagement.md

```diff
@@ -0,0 +1,31 @@
+# Extract secrets from config files or use NPM package that encrypts them
+
+### One Paragraph Explainer
+The most common and secure way to provide a Node.js application access to keys and secrets is to store them using environment variables on the system where it is being run. Once set, these can be accessed from the global `process.env` object. 
+A litmus test for whether an app has all config correctly factored out of the code is whether the codebase could be made open source at any moment, without compromising any credentials.
+
+For rare situations where secrets do need to be stored inside source control, using a package such as [cryptr](https://www.npmjs.com/package/cryptr) allows these to be stored in an encrypted form as opposed to in plain text.
```

Comment:
@alonbardavid Welcome!

We all agree (do we?) that using a vault is the prominent option. Alternatives can be considered to small projects, is git encryption a valid option for small project?

To my & @js-kyle understanding it's a valid option - help us realize why not: the decryption-key can be shared via secured channel (e.g. a phsical note), it can be shared only with few developers or with devops only, it's better than nothing. What do you think?

@js-kyle https://github.com/shadowhand/git-encrypt seems more popular

Reply:
@alonbardavid I think I see what you're saying - that if the option is available for the decryption key to be provided at runtime , why wouldn't that method be used used to store the secret in the first place?

@i0natan I chose cryptr as I felt this article was becoming a bit too broad, git-encrypt seems less Node.js focussed, what do you think?

Reply:
@js-kyle precisely. Especially since you'd want the key to the secret to have enough entropy to not be breakable on it's own, which usually means it'll be hard to remember. If it's hard to remember, you are going to put it in a password safe anyway, might as well put the actual secret there.